databricks/koalas/plot.py

```diff
@@ -1118,6 +1120,103 @@ def _plot(data, x=None, y=None, subplots=False, ax=None, kind="line", **kwds):
     return plot_obj.result
 
 
+def _find_backend(backend):
+    """
+    Find a koalas plotting backend
+    """
+    # function copied from pandas.plotting._core
+
+    # Delay import for performance.
+    import pkg_resources
+
+    for entry_point in pkg_resources.iter_entry_points("koalas_plotting_backends"):
+        if entry_point.name == "matplotlib":
+            # matplotlib is an optional dependency. When
+            # missing, this would raise.
+            continue
+        _backends[entry_point.name] = entry_point.load()
+
+    try:
+        return _backends[backend]
+    except KeyError:
+        try:
+            module = importlib.import_module(backend)
+        except ImportError:
+            # We re-raise later on.
+            pass
+        else:
+            if hasattr(module, "plot"):
+                # Validate that the interface is implemented when the option
+                # is set, rather than at plot time.
+                _backends[backend] = module
+                return module
+
+    # if failed to load plotting backend, set it back to default
+    from databricks.koalas import set_option
+    set_option("plotting.backend", "matplotlib")
+
+    raise ValueError(
+        "Could not find plotting backend '{backend}'. Ensure that you've installed "
+        "the package providing the '{backend}' entrypoint, or that the package has a "
+        "top-level `.plot` method.".format(backend=backend)
+    )
+
+
+def _get_plot_backend(backend=None):
+    # function copied from pandas.plotting._core
+
+    backend = backend or get_option("plotting.backend")
+
+    if backend == "matplotlib":
+        # Because matplotlib is an optional dependency and first-party backend,
+        # we need to attempt an import here to raise an ImportError if needed.
+        try:
+            import databricks.koalas.plot as module
+        except ImportError:
+            raise ImportError(
+                "matplotlib is required for plotting when the "
+                "default backend 'matplotlib' is selected."
+            ) from None
+
+        _backends["matplotlib"] = module
+
+    if backend in _backends:
+        return _backends[backend]
+
+    module = _find_backend(backend)
+    _backends[backend] = module
+    return module
+
+    def _get_args_map(backend_name, data, kind, kwargs):
```

Comment:
indent?

Reply:
Is there a problem with indentation here?

Reply:
Now that this function is top level, so we have to not indent this function.